R/link_ch_with_ae.R

```diff
@@ -0,0 +1,119 @@
+#' Link AE episodes released from CH
+#'
+#' @description 1. Add a variable, sc_ch_link_ae,
+#' for CH episode to indicate discharge to AE2.
+#' 2. Populate ch_name and ch_postcode to relavent AE episodes from CH episodes
+#'
+#' @param ep episode file
+link_ch_with_ae <- function(episode) {
+  ep <- ep %>%
+    dplyr::mutate(ep_row_id_CE = dplyr::row_number())
+
+  data <- ep %>%
+    dplyr::select(
+      ep_row_id_CE,
+      anon_chi,
+      recid,
+      record_keydate1,
+      record_keydate2,
+      keytime1,
+      keytime2,
+      ch_name,
+      ch_postcode,
+      cup_pathway,
+      cij_pattype,
+      cij_ipdc,
+      cij_start_date
+    )
+
+  # pull care home data
+  care_home_data <- data %>%
+    dplyr::filter(recid == "CH") %>%
+    dplyr::arrange(anon_chi, record_keydate1)
+
+  # populate ch_name and ch_postcode from care home to ae episodes ...
+  # ... if someone goes to ae from ch
+  adms1 <- data %>%
+    # c("01B", "04B", "GLS", "AE2") are included to link with CH.
+    # Technically, DD episodes can also be linked with CH episodes ...
+    # ... as DD episodes are linked with AE episodes and others.
+    dplyr::filter(recid %in% c("01B", "04B", "GLS", "AE2")) %>%
+    # remove A&E records that don't lead to an admission,
+    # remove elective admissions,
+    # and remove day cases
+    dplyr::filter(recid == "AE2" &
+      grepl("A", cup_pathway) == TRUE |
+      recid != "AE2") %>%
+    dplyr::filter(!(recid != "AE2" &
+      cij_pattype == "Elective")) %>%
+    dplyr::filter(!(recid != "AE2" &
+      cij_ipdc == "D")) %>%
```

Comment:
We will review this code as this was initially some code used that was specific to the query LIST had.